src/lib/index.js

```diff
@@ -344,6 +344,17 @@ lib.mergeArray = function(traceAttr, cd, cdAttr) {
     }
 };
 
+lib.getTargetArray = function(trace, target) {
+    if(typeof target === 'string' && target) {
+        var array = lib.nestedProperty(trace, target).get();
+
+        return Array.isArray(array) ? array : [];
```

Comment:
Either this should be `: false` or [the one below](https://github.com/plotly/plotly.js/pull/1609/files/0be666f1c94d6762561c09aca3c9965ae09f53fa..823375ccbdf6260c5b7278cbdd4a62908da6ff1c#diff-e64ef427412e9bab4929b8b93ca12d96R355) should be `[]`

I'd probably vote for `false` and then instead of the `if(!len) return;` check, do `if(!targetArray) return;`

That said, this raises the question, in these applications, of what we should do when `targetArray` isn't the same length as the other associated arrays. Looks like for both, it will truncate (or extend) all of them to the length of `targetArray`. Is that what we want? Seems right for sort, but if we agree then lets 🔒  it with a test. For filter there's an argument to be made that eg `[]` and `)(`, and `{}` and `}{`, should be complementary - which they are *within* `targetArray` but not with respect to the items *beyond* it. Maybe that's still what we want, or maybe we don't want the extra complexity in so far as we think this would be incorrect usage with no legitimate use case. I don't have a strong opinion other than that it should be clear how we handle this case.

BTW it took me a minute to recall the motivation for the different cases in this function - can you give the function a docstring?